src/bin/nydusd/fusedev.rs

```diff
@@ -175,14 +176,20 @@ impl FusedevDaemon {
         let thread = thread::Builder::new()
             .name("fuse_server".to_string())
             .spawn(move || {
-                let _ = s.svc_loop(&inflight_op);
-                // quit the daemon if any fuse server thread exits
-                exit_daemon();
+                if let Err(err) = s.svc_loop(&inflight_op) {
+                    warn!("fuse server exit with err: {:?}, exit daemon", err);
+                    if let Err(e) = thread::Builder::new()
+                        .name("exit_thread".to_string())
+                        .spawn(exit_daemon)
```

Comment:
Why `exit_daemon` has to be invoked in another thread? A possible reason is state machine now has blocking operation inside?

Reply:
> Why `exit_daemon` has to be invoked in another thread? A possible reason is state machine now has blocking operation inside?

`A possible reason is state machine now has blocking operation inside` That is the point.  As a result, thread exiting is decoupling with process exiting by new thread.